### PR TITLE
Do not return secret token in the Oauth API

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
* Remove the `personalAccessTokenManager.get()` call from the OAuth API `getToken()` method. The OAuth API must not know anything about PAT secrets. It should get tokens only by requesting an SCM provider OAuth API.
* Fix validating the Bitbucket-Server PAT method by requesting user instead of requesting.

This prevents the code execution going to a recursive loop: `bitbucketServerApiClient.getPersonalAccessToken()` calls `oauthApi.getToken()` which referred to `personalAccessTokenManager.getToken()` which validated the token by calling `scmPersonalAccessTokenFetcher.getScmUsername()` -> `bitbucketServerApiClient.getPersonalAccessToken()`.

### What issues does this PR fix or reference?

https://issues.redhat.com/browse/CRW-4351

### How to test this PR?

1. Setup [bitbucket-server oauth](https://eclipse.dev/che/docs/stable/administration-guide/configuring-oauth-2-for-a-bitbucket-server/)
2. Start a workspace from the bitbucket-server repo to initiate the OAuth exchange.
3. Go to dashboard -> User preferences -> Personal Access Tokens tab and add a new bitbucket-server token. You can use a random string for the token input.

See: the token creation is rejected, but no backend errors appear.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/EmbeddedOAuthAPI.java [hunk 1]
```diff
@@ -159,25 +153,11 @@ public OAuthToken getToken(String oauthProvider)
...
       }
-      Optional<PersonalAccessToken> tokenOptional =
-          personalAccessTokenManager.get(subject, provider.getEndpointUrl());
-      if (tokenOptional.isPresent()) {
-        PersonalAccessToken accessToken = tokenOptional.get();
-        return newDto(OAuthToken.class).withToken(accessToken.getToken());
-      }
       throw new UnauthorizedException(
           "OAuth token for user " + subject.getUserId() + " was not found");
-    } catch (IOException | ScmConfigurationPersistenceException | ScmCommunicationException e) {
```
Comment (tolusha) [inline wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/EmbeddedOAuthAPI.java:170]: That snippet above was add as a part of https://github.com/eclipse-che/che-server/pull/408
Could you check if that functionality still works?

Comment (vinokurig) [reply wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/EmbeddedOAuthAPI.java:170]: Yes, I have checked the revoke token functionality and it works fine.

File: wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/EmbeddedOAuthAPI.java [hunk 2]
```diff
@@ -159,25 +153,11 @@ public OAuthToken getToken(String oauthProvider)
       if (token != null) {
         return token;
       }
-      Optional<PersonalAccessToken> tokenOptional =
-          personalAccessTokenManager.get(subject, provider.getEndpointUrl());
-      if (tokenOptional.isPresent()) {
-        PersonalAccessToken accessToken = tokenOptional.get();
-        return newDto(OAuthToken.class).withToken(accessToken.getToken());
-      }
```
Comment (ibuziuk) [inline wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/EmbeddedOAuthAPI.java:167]: @vinokurig why have we got PAT as part of OAuth flow in general and now removing it?

Comment (vinokurig) [reply wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/EmbeddedOAuthAPI.java:167]: I believe this was added by a mistake.

**Review by tolusha — APPROVED** [review]

**Review by dmytro-ndp — APPROVED** [review]
I was able to continue working with Eclipse Che after executing of step 3:
> 3. Che Setup [bitbucket-server oauth](https://eclipse.dev/che/docs/stable/administration-guide/configuring-oauth-2-for-a-bitbucket-server/)
Start a workspace from the bitbucket-server repo to initiate the OAuth exchange.
Go to dashboard -> User preferences -> Personal Access Tokens tab and add a new bitbucket-server token. You can use a random string for the token input.

Screenshot from 2023-08-15 18-03-03

There was no "backend is not available" error [1] observed.

@vinokurig : could you please prepare backport PR to 7.72.x branch for review to make the fix available in DS 3.8.0?

[1] https://issues.redhat.com/browse/CRW-4351

**Pull request merged**